darkdraw/save_ans.py

```diff
@@ -0,0 +1,160 @@
+    parts = color_str.split(' on ')
+    if len(parts) == 1:
+        fg_part = parts[0]
+        bg = None
+    else:
+        fg_part = parts[0]
+        bg = parts[1].strip()
+
+    words = fg_part.split()
+    attributes = []
+    color_candidates = []
+    for word in words:
+        if word in ['bold', 'italic', 'underline', 'reverse', 'dim', 'blink']:
+            attributes.append(word)
+        else:
+            color_candidates.append(word)
+
+    if color_candidates:
+        for candidate in color_candidates:
+            if candidate.isdigit():
+                fg = candidate
+                break
+        else:
+            fg = color_candidates[-1]
+    else:
+        fg = 'white'
+
+    return attributes, fg, bg
+
+def get_escape_codes(attributes, fg, bg):
+    """Generate ANSI escape codes for attributes and colors."""
+    codes = []
+    attr_codes = {
+        'bold': '1', 'dim': '2', 'italic': '3',
+        'underline': '4', 'blink': '5', 'reverse': '7',
+    }
+    for attr in attributes:
+        if attr in attr_codes:
+            codes.append(attr_codes[attr])
+
+    fg_code = get_color_code(fg)
+    codes.append(f'38;5;{fg_code}')
+
+    if bg:
+        bg_code = get_color_code(bg)
+        codes.append(f'48;5;{bg_code}')
+
+    return '\033[' + ';'.join(codes) + 'm' if codes else ''
+
+def export_frame(rows, x_col, y_col, text_col, color_col):
+    """Export a single frame to ANSI text."""
+    if not rows:
+        return ''
+
+    # Find max_x and max_y
+    max_x = 0
+    max_y = 0
+    for row in rows:
+        x = int(x_col.getValue(row))
+        y = int(y_col.getValue(row))
+        text = str(text_col.getValue(row))
+        max_x = max(max_x, x + len(text) - 1)
+        max_y = max(max_y, y)
+
+    # Create grid
+    grid = [[None] * (max_x + 1) for _ in range(max_y + 1)]
+
+    # Place characters
+    for row in rows:
+        x = int(x_col.getValue(row))
+        y = int(y_col.getValue(row))
+        text = str(text_col.getValue(row))
+        color_str = str(color_col.getValue(row))
+        attributes, fg, bg = parse_color(color_str)
+        for i, char in enumerate(text):
+            grid[y][x + i] = (char, attributes, fg, bg)
+
+    # Generate output
+    output = ''
+    for y in range(max_y + 1):
+        line = ''
+        prev_codes = None
+        for x in range(max_x + 1):
+            cell = grid[y][x]
+            if cell:
+                char, attributes, fg, bg = cell
+                codes = get_escape_codes(attributes, fg, bg)
+                if codes != prev_codes:
+                    line += codes
+                    prev_codes = codes
+                line += char
+            else:
+                if prev_codes != '\033[0m':
+                    line += '\033[0m'
+                    prev_codes = '\033[0m'
+                line += ' '
+        line += '\033[0m'
+        output += line + '\n'
+    return output
+
+@VisiData.api
+def save_ans(vd, p, sheet):
+    output_filename = p
```

Comment:
This line is not needed as it is duplicated on line 157.  (For my money, you could just pass `p` into `open()`, and remove both).  This is particularly salient here since the docstring should by convention be the first line after the `def`.